Userland/Services/SpiceAgent/SpiceAgent.cpp

```diff
@@ -112,11 +113,31 @@ ErrorOr<void> SpiceAgent::on_clipboard_message(ClipboardMessage& message)
     case ClipboardDataType::Text: {
         // The default mime_type for set_data is `text/plain`
         GUI::Clipboard::the().set_data(message.contents());
-        return {};
+        break;
+    }
+
+    // For the image formats, let's try to find a decoder from LibGfx
+    case ClipboardDataType::PNG:
+    case ClipboardDataType::BMP:
+    case ClipboardDataType::JPG:
+    case ClipboardDataType::TIFF: {
+        auto mime_type = TRY(to_mime_type(message.data_type()));
+        auto decoder = Gfx::ImageDecoder::try_create_for_raw_bytes(message.contents(), mime_type.to_deprecated_string());
```

Comment:
Yikes, this helper wants an actual String rather than a StringView for mime type detection? :( Big ol todo if that's the case,.

Reply:
Shouldn't be too difficult to make it take a StringView in this PR

Reply:
I'll put a FIXME